configcatclient/lazyloadingcachepolicy.py

```diff
@@ -43,22 +43,25 @@ def get(self):
 
     def force_refresh(self):
         try:
+            self._lock.acquire_write()
+            # If while waiting to acquire the write lock another
+            # thread has updated the content, then don't bother requesting
+            # to the server to minimise time.
+            if self._last_updated is not None and self._last_updated + self._cache_time_to_live >= datetime.datetime.utcnow():
```

Comment:
We would like to guarantee that the force_refresh will update the config regardless of the cache TTL.
Can I ask what is your use-case? How do you call force_refresh() from your code?

I think a solution can be if you inherit from this class and add this functionality to the new class.

Reply:
So a bit of context, and let me know if I don't explain it so well.
What we have is a process that has many threads, say 10 threads.
We have many threads that can serve a request at the same time, and all of those may be from different users.
They all check if a specific flag is enabled.
So what happens is you'll have 10 threads all hit the TTL/cache timeout at the same time and then proceed to force refresh at the same time. Now in theory one thread is enough to update the cache and also TTL, but each one of those other threads will then do a request to the servers, takings around 400-500ms and then hit the is_fetched() condition and basically all update TTL/_last_updated
The aim here is to try and avoid that scenario where you have those other threads doing unnecessary work/requests, as a simple datetime check is far quicker than any network request.

So how about a middle ground, where force_refresh is always going to update the cache, but moving the write lock outside of force_refresh and having this early exit within the get call?
https://github.com/configcat/python-sdk/blob/master/configcatclient/lazyloadingcachepolicy.py#L34

I'll put up a patch as an example, and you can let me know from that.